docs/usage.md

```diff
@@ -0,0 +1,309 @@
+  with:
+    distribution: '<distribution>'
+    java-version: '11.x'
+    architecture: x86 # optional - defaults to x64
+- run: java -cp java HelloWorldApp
+```
+
+## Installing Java from local file
+If your use-case requires using custom distribution or version that is not provided in action by default, you can download it manually and then action will take care about installation and caching version on VM:
+
+TO-DO: Fix example
+```yaml
+steps:
+- run: |
+    download_url="https://github.com/AdoptOpenJDK/openjdk11-binaries/releases/download/jdk-11.0.10%2B9/OpenJDK11U-jdk_x64_mac_hotspot_11.0.10_9.pkg"
+    wget $download_url
+- uses: actions/setup-java@v2-preview
+  with:
+    distribution: 'jdkFile'
+    jdkFile: runner.temp # Optional - jdkFile to install java from. Useful for distributions not provided by default
+    java-version: '11.0.0'
+    architecture: x64
+    
+- run: java -cp java HelloWorldApp
+```
+
+## Testing against different Java versions
+```yaml
+jobs:
+  build:
+    runs-on: ubuntu-20.04
+    strategy:
+      matrix:
+        java: [ '8', '11', '13', '15' ]
+    name: Java ${{ matrix.Java }} sample
+    steps:
+      - uses: actions/checkout@v2
+      - name: Setup java
+        uses: actions/setup-java@v2-preview
+        with:
+          distribution: '<distribution>'
+          java-version: ${{ matrix.java }}
+      - run: java -cp java HelloWorldApp
+```
+
+## Testing against different Java distributions
+**NOTE:** Please pay attention that different distributors provide different list of available versions / supported configurations.
+```yaml
+jobs:
+  build:
+    runs-on: ubuntu-20.04
+    strategy:
+      matrix:
+        distribution: [ 'zulu', 'adoptium' ]
+        java: [ '8', '11', '13', '15' ]
+    name: Java ${{ matrix.Java }} (${{ matrix.distribution }}) sample
+    steps:
+      - uses: actions/checkout@v2
+      - name: Setup java
+        uses: actions/setup-java@v2-preview
+        with:
+          distribution: ${{ matrix.distribution }}
+          java-version: ${{ matrix.java }}
+      - run: java -cp java HelloWorldApp
+```
+
+## Publishing using Apache Maven
+
+### Extra setup for pom.xml:
+
+The Maven GPG Plugin configuration in the pom.xml file should contain the following structure to avoid possible issues like `Inappropriate ioctl for device` or `gpg: signing failed: No such file or directory`:
+
+```xml
+<configuration>
+  <!-- Prevent gpg from using pinentry programs -->
+  <gpgArguments>
+    <arg>--pinentry-mode</arg>
+    <arg>loopback</arg>
+  </gpgArguments>
+</configuration>
+```
+GPG 2.1 requires `--pinentry-mode` to be set to `loopback` in order to pick up the `gpg.passphrase` value defined in Maven `settings.xml`.
+
+### Yaml example:
+```yaml
+jobs:
+  build:
+
+    runs-on: ubuntu-latest
+
+    steps:
+    - uses: actions/checkout@v2
+    - name: Set up JDK 8.x
+      uses: actions/setup-java@v2-preview
+      with:
+        distribution: 'zulu'
```

Comment:
in all examples we use `'<distribution>'` but `'zulu'` here. Probably makes sense update it to `'<distribution>'` too